src/main/java/edu/umich/its/cpm/EmailFormatter.java

```diff
@@ -41,58 +58,189 @@ public EmailFormatter(String emailMgs, AttachmentHandler attachmentHandler) thro
+            status.remove(Utils.JSON_ATTR_MAIL_ATTACHMENTS);
+            emailMsgPlusStatus.set(1, status);
+            return emailMsgPlusStatus;
+        }
+        // Large email size check, if grater than proposed limit attachment will be dropped
+        if (checkMsgSizeMoreThanExpectedLimit(rfcFormattedText)) {
+            log.warn("Attachments are dropped for the message");
+
+            rfcFormattedText = removeAttachments(rfcFormattedText);
+            emailMsgPlusStatus.set(0, rfcFormattedText);
+            status.put(Utils.JSON_ATTR_ITEM_STATUS, Utils.STATUS_PARTIAL);
+            status.put(Utils.JSON_ATTR_MESSAGE, "email Message size exceeds the expected limit, attachments dropped");
+            status.remove(Utils.JSON_ATTR_MAIL_ATTACHMENTS);
+            emailMsgPlusStatus.set(1, status);
+
+            return emailMsgPlusStatus;
+        }
+
+        JSONArray attachments = status.getJSONArray("attachments");
+        for (int i = 0; i < attachments.length(); i++) {
+            JSONObject attach = attachments.getJSONObject(i);
+            if(attach.get(Utils.MIGRATION_STATUS)!= Utils.STATUS_OK){
+                status.put(Utils.JSON_ATTR_ITEM_STATUS, Utils.STATUS_PARTIAL);
+                status.put(Utils.JSON_ATTR_MESSAGE,"attachments URL from Ctools couldn't be resolved, some or all of the attachments dropped");
+                status.remove(Utils.JSON_ATTR_MAIL_ATTACHMENTS);
+                emailMsgPlusStatus.set(1, status);
+                return emailMsgPlusStatus;
+            }
+        }
+        // everything went fine at this point
+        status.put(Utils.JSON_ATTR_ITEM_STATUS, Utils.STATUS_OK);
+        status.put(Utils.JSON_ATTR_MESSAGE, Utils.SUCCESS_MSG);
+        status.remove(Utils.JSON_ATTR_MAIL_ATTACHMENTS);
+        emailMsgPlusStatus.set(1, status);
+        return emailMsgPlusStatus;
     }
 
-    public String rfc822Format() {
+    public List<Object> rfcFormatWithoutAttachmentLimitCheck() {
         ArrayList<String> headers = prunedHeadersWithoutContentType();
-        return getFormattedEmailText(headers).toString();
+        List<Object> formattedEmailText = getFormattedEmailText(headers);
+        return formattedEmailText;
+    }
+
+    public String removeAttachments(String rfcFormattedText) {
+        String emailWithOutattachments = subStringToRemoveAttachments(rfcFormattedText);
+        return emailWithOutattachments;
+    }
+
+    private String subStringToRemoveAttachments(String rfcFormattedText) {
+        StringBuilder emailTextWithBodyNoAttachments = new StringBuilder();
+        String attachmentRemovedText = subStringBefore(rfcFormattedText, "Content-Transfer-Encoding: base64");
+        String[] msgBodyWithSomeExtra = attachmentRemovedText.split(NEW_LINE);
+        List<String> emailTextList = new LinkedList(Arrays.asList(msgBodyWithSomeExtra));
+        int size = emailTextList.size();
+        emailTextList.remove(size - 1); // remove the last element in the list
```

Comment:
So here is a email stream, the bolded is the last element in the list and this is part of  attachment item so i am removing it and also marking the line above as a boundary by adding -- as end of the line message. 

`MIME-Version: 1.0
Received: by 10.202.86.196 with HTTP; Wed, 24 Aug 2016 07:04:47 -0700 (PDT)
From: Jane Doe <jdoe@umich.edu>
Date: Wed, 24 Aug 2016 10:04:47 -0400
Message-ID: <CAF8WsuqBzODBcc5-d_6yEgoyQhi13+bpLHjMNWP=g088YR=-5w@mail.gmail.com>
Subject: I figured it out
To: pushyami-test@ctqa.dsc.umich.edu
List-Id: <main.localhost>
Content-Type: multipart/mixed; 
	boundary="----=_Part_0_1587819720.1475501950578"

------=_Part_0_1587819720.1475501950578
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 7bit

This is so awesome to find out and unbelievable
------=_Part_0_1587819720.1475501950578
**Content-Type: text/plain; charset=UTF-8; name=email_msg.txt**
`